File: 94-binary-tree-inorder-traversal/binary-tree-inorder-traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var inorderTraversal = function(root) {
    let arr = [];
    Helper(root, arr);
    return arr;
};

function Helper(root, arr) {
    if(root === null) {
        return
    }
    Helper(root.left, arr);
    arr.push(root.val);
    Helper(root.right, arr)
}

File: 94-binary-tree-inorder-traversal/binary-tree-inorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        result = []
        self.Helper(root, result)
        return result
    def Helper(self,node, ans):
        if node is None:
            return
        self.Helper(node.left, ans)
        ans.append(node.val)
        self.Helper(node.right, ans)


    
        